sql/updates/world/3.3.5/2016_08_08_4001_world_335.sql

```diff
@@ -0,0 +1,60 @@
+-- NPC ID 9021 Kharan Mighthammer, Quest ID 4001 'What Is Going On?' and 4342 'Kharan's Tale'
+SET @Kharan := 9021;
+
+UPDATE `creature_template` SET `AIName`= 'SmartAI', `ScriptName`= '' WHERE `entry` = @Kharan;
+
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = @Kharan);
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@Kharan,0, 0,0, 10,0,100,0,   0,20,0,0,  1,   0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Kharan Mighthammer - Within 0-20 Range Out of Combat LoS - Say Line 0'),
+(@Kharan,0, 1,2, 62,0,100,0,1823, 0,0,0, 72,   0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Kharan Mighthammer - on Gossip option 0 selected - Close Gossip'),
+(@Kharan,0, 2,0, 61,0,100,0,   0, 0,0,0, 15,4342,0,0,0,0,0,7,0,0,0,0,0,0,0,"Kharan Mighthammer - on Gossip option 0 selected - Quest Credit 'Kharan's Tale'"),
+(@Kharan,0, 3,4, 62,0,100,0,1839, 0,0,0, 72,   0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Kharan Mighthammer - on Gossip option 0 selected - Close Gossip'),
+(@Kharan,0, 4,0, 61,0,100,0,   0, 0,0,0, 15,4001,0,0,0,0,0,7,0,0,0,0,0,0,0,"Kharan Mighthammer - on Gossip option 0 selected - Credit Quest 'What Is Going On?'");
+
+DELETE FROM `creature_text` WHERE `entry` = @Kharan;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
+(@Kharan,0,0,'HEY! HEY YOU! $R! Get me out of here!',12,0,100,0,0,0,4725,0,'Kharan Mighthammer');
```

Comment:
Oh. OK, right. I noticed that those texts were located near each other in `broadcast_text`, but I thought it was because they were used by the other nearby prisoners. Very well, I will get to it and create a new commit for it. Thanks for the suggestion!